### PR TITLE
fix(DSC): when DSC is in Error because of missing DSCI CR

### PR DESCRIPTION
**Pull request by zdtsw**
- DSCI creation should trigger watches() and reconcile DSC

could have the logic in the same `watchDataScienceClusterResources`()
but think it would be more clear to have one dedicated for DSCI case

## How Has This Been Tested?

before
- install operator
- create DSC CR(d not create DSCI CR)
- DSC CR phase is Error
- create DSCI CR
- nothing happened
- delete DSC CR
- re-create DSC CR
- start to reconcile Managed Components

now
- install operator
- create DSC CR (do not create DSCI)
- DSC CR got phase "Error" and no component should be deployed regardless ManagementStates
- Create DSCI CR
- DSC CR start to deploy Managed Component when all in ReconcileCompleted it gets Ready as phase
- 

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: controllers/datasciencecluster/datasciencecluster_controller.go [hunk 1]
```diff
@@ -435,6 +435,29 @@ func (r *DataScienceClusterReconciler) SetupWithManager(mgr ctrl.Manager) error
...
+	switch {
+	case len(instanceList.Items) == 1:
+		requestName = instanceList.Items[0].Name
+	case len(instanceList.Items) == 0:
+		requestName = "default-dsc"
+	default:
+		return nil
+	}
+	// When DSCI CR gets created, trigger reconcile function
+	if a.GetObjectKind().GroupVersionKind().Kind == "DSCInitialization" || a.GetName() == "default-dsci" {
```
Comment (etirelli) [inline controllers/datasciencecluster/datasciencecluster_controller.go:454]: just out of curiosity, why do you need to check for ```|| a.GetName() == "default-dsci"```? what is the scenario where that would trigger?

**Review by etirelli — APPROVED** [review]
LGTM

File: controllers/datasciencecluster/datasciencecluster_controller.go (hunk 1 above)
Comment (zdtsw) [reply controllers/datasciencecluster/datasciencecluster_controller.go:454]: I first used the check only on the` .Kind`, but i think in some cases it returns empty string, this also happens on `.Group`
my understanding it actually comes from https://github.com/kubernetes-sigs/controller-runtime/issues/1517
thats why we usually client.Object.GetName() in most of our code base. in theory, in ODH user can create their  DSCI CR with any name they want, in downstream, till we have webhook in place, default-dsci will the only valid name for DSCI CR.

Comment (etirelli) [reply controllers/datasciencecluster/datasciencecluster_controller.go:454]: thank you. We might need to find an alternative to checking for the name directly, but good enough for now.

**Pull request merged**